### PR TITLE
feat: add `space` example

### PR DESCRIPTION
**Pull request by JayWhite2357**
# Rationale for this change

More examples in the repo provide better documentation as well as provide some regression testing on top of other tests.

# What changes are included in this PR?

An example (related to space) of reading tables for a csv file is provided along with various queries against those tables.

# Are these changes tested?
Yes

### CODE REVIEW

File: crates/proof-of-sql/Cargo.toml [hunk 1]
```diff
@@ -75,7 +75,8 @@ default = ["arrow", "perf"]
 arrow = ["dep:arrow", "std"]
 blitzar = ["dep:blitzar", "dep:merlin", "std"]
 test = ["dep:rand", "std"]
-perf = ["blitzar", "rayon", "ark-ec/parallel", "ark-poly/parallel", "ark-ff/asm"]
+perf = ["blitzar", "cpu-perf"]
+cpu-perf = ["rayon", "ark-ec/parallel", "ark-poly/parallel", "ark-ff/asm"]
```
Comment (JayWhite2357) [inline crates/proof-of-sql/Cargo.toml:79]: This isn't directly related to this example. I just added it here as a convenience going forward.

Comment (Dustin-Ray) [reply crates/proof-of-sql/Cargo.toml:79]: Looks good thank you

File: crates/proof-of-sql/examples/space/main.rs [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+// We generate the public parameters and the setups used by the prover and verifier for the Dory PCS.
+// The `max_nu` should be set such that the maximum table size is less than `2^(2*max_nu-1)`.
+// For a sampling:
+// max_nu = 3 => max table size is 32 rows
+// max_nu = 4 => max table size is 128 rows
+// max_nu = 8 => max table size is 32768 rows
+// max_nu = 10 => max table size is 0.5 million rows
+// max_nu = 15 => max table size is 0.5 billion rows
+// max_nu = 20 => max table size is 0.5 trillion rows
+// Note: we will eventually load these from a file.
```
Comment (Dustin-Ray) [inline crates/proof-of-sql/examples/space/main.rs:33]: I like these comments a lot

**Review by Dustin-Ray — APPROVED** [review]
Conditionally approved once CI passes rebase, thanks!

**Review by iajoiner — APPROVED** [review]

**Pull request merged**